wikimedia/wikipedia-ios
# Create account username verification feedback

**Phabricator:** https://phabricator.wikimedia.org/T347999

### Notes
This PR makes a few tweaks to the original fix in https://github.com/wikimedia/wikipedia-ios/pull/4647.

1. I removed the Task and await syntax, mainly because the `accountCreator.checkUsername` method was already designed in the old concurrency style with completion and failure blocks. To me, making it more consistent with the old style felt less confusing.
2. I noticed that it was making the username check API call twice in rapid succession, so I added a boolean to guard against that.
3. Applied design review feedback (adjusted spacing, ensured the correct red color is used).

### Test Steps
1. Log out, go to create account screen.
2. Type "1.2.3.4" in the username field.
3. When field loses focus, you should see a "username not available" error message appear.
4. Type a username that is available.
5. When field loses focus, you should not see an error message.
